packages/protocol/contracts/lib/Checkpointing.sol

```diff
@@ -90,17 +93,20 @@ library Checkpointing {
      * @param _value Numeric value to be registered at the given point in time
      */
     function _add192(History storage self, uint64 _time, uint192 _value) private {
-        uint256 length = self.history.length;
-        if (length == 0 || self.history[self.history.length - 1].time < _time) {
-            // If there was no value registered or the given point in time is after the latest registered value,
-            // we can insert it to the history directly.
```

Comment:
Maybe we can keep (or replace) these comments to help understanding the logic.

Reply:
Agreed, do you mind doing it? I didn't do it yet because the logic changed slightly :).